src/main/kotlin/org/jetbrains/teamcity/rest/implementation.kt

```diff
@@ -66,18 +68,24 @@ internal class TeamCityInstanceImpl(private val serverUrl: String,
 
 private class BuildLocatorImpl(private val service: TeamCityService, private val serverUrl: String): BuildLocator {
     private var buildConfigurationId: BuildConfigurationId? = null
+    private var number: String? = null
     private var status: BuildStatus? = BuildStatus.SUCCESS
     private var tags = ArrayList<String>()
     private var count: Int? = null
     private var branch: String? = null
     private var includeAllBranches = false
     private var pinnedOnly = false
 
-    override fun fromConfiguration(buildConfigurationId: BuildConfigurationId): BuildLocator {
+    override fun fromConfiguration(buildConfigurationId: BuildConfigurationId): BuildLocatorImpl {
```

Comment:
So it would be better to improve the test to avoid such problems. It isn't good if a test depend on external environment, but if it is, we can at least write code in the test which will either repeat the test if network resources isn't available, or skip it using [assume API](http://junit.org/junit4/javadoc/4.12/org/junit/Assume.html).